src/background/zkKeeper.ts

```diff
@@ -37,6 +37,22 @@ export default class ZkKeeperController extends Handler {
     this.approvalService = new ApprovalService();
     this.walletService = new WalletService();
     this.lockService = LockService.getInstance();
+
+    BrowserUtils.addRemoveWindowListener((windowId: number) => {
+      log.debug("Inside removeWindow onRemove");
+
+      try {
+        log.debug("Inside removeWindow onRemove locked");
+        if (BrowserUtils.cached?.id === windowId) {
+          BrowserUtils.cached = null;
```

Comment:
I don't think it's a good idea to change cached value outside `BrowserUtils` class.
I'd move this logic inside the class.